NativeApp/src/components/Home/container.js

```diff
@@ -18,20 +21,112 @@ export default Container => class extends Component {
       super(props);
       this.state = {
         takenHolidays: {},
+        showModal: false,
+        user: null,
+        booking: {
+          startDate: '',
+          endDate: '',
+          halfDay: false,
+        },
       };
     }
 
     componentDidMount() {
       getTakenHolidays()
         .then(data => this.setState({ takenHolidays: this.formatDate(data) }));
+
+      userProfile()
+        .then(user => this.setState({ user }));
+    }
+
+    onDayPress = (day) => {
+      if (day) {
+        this.setState({
+          showModal: true,
+          booking: {
+            startDate: day.dateString,
+            endDate: day.dateString,
+          },
+        });
+      }
+    }
+
+    submitRequest = () => {
+      const { booking, user } = this.state;
+      const request = {
+        dates: [
+          {
+            endDate: booking.endDate,
+            halfDay: false,
+            startDate: booking.startDate,
+          },
+        ],
+        employeeId: user.employeeId,
+      };
+
+      requestHolidays(request)
+        .then(() => {
+          getTakenHolidays()
+            .then(data => this.setState({ takenHolidays: this.formatDate(data) }));
+          this.closeModal();
+        })
+        .catch(e => Alert.alert(
+          'Could not request holidays',
+          e.message,
+        ));
+    }
+
+    changeStartDate = (date) => {
+      const formatDate = moment(date).format('YYYY-MM-DD');
+      this.setState(prevState => ({
+        booking: {
+          ...prevState.booking,
+          startDate: formatDate,
+        },
+      }));
+    }
+
+    changeEndDate = (endDate) => {
+      const formatEndDate = moment(endDate).format('YYYY-MM-DD');
+      this.setState(prevState => ({
+        booking: {
+          ...prevState.booking,
+          endDate: formatEndDate,
+        },
+      }));
+    }
+
+    closeModal = () => {
+      const { showModal } = this.state;
+      this.setState({ showModal: !showModal });
+    }
+
+    enumerateDaysBetweenDates = (startDate, endDate) => {
+      const dates = [startDate];
+
+      const currDate = moment(startDate).startOf('day');
+      const lastDate = moment(endDate).startOf('day');
+
+
+      while (currDate.add(1, 'days').diff(lastDate) < 0) {
+        dates.push(currDate.clone().format('YYYY-MM-DD'));
+      }
+
+      dates.push(endDate);
+
+      return dates;
     }
 
     formatDate = data => data.reduce((obj, item) => {
       const holidayStatus = this.holidayStatus(item.holidayStatusId);
-      obj[item.start] = { textColor: 'white', color: holidayStatus };
+      const dates = this.enumerateDaysBetweenDates(item.start, item.end);
+      for (let i = 0; i < dates.length; i += 1) {
```

Comment:
could this not be done as a foreach loop? much nicer syntax

Reply:
Can't see why it couldn't.